### DISCUSSION TITLE
Best way to apply updates to a record one column at a time with dynamic column name?

### DISCUSSION THREAD

**Discussion by mattsimgithub** [simolus3/drift#2228] [Q&A] [answered]
I have an application that collects "audits". An audit is mostly a series of pass/fail questions, with an optional comment if the question was marked "fail".

The table for an audit has a boolean column for each question type and a string column for the optional comment.

I want to update the audit record one column at a time as input comes in from the user. I'd like to do this using just the column name (this makes other parts of the app simpler). Presently I have this:

```dart
  Future<bool> updateAudit(String auditId, String questionKey, dynamic newValue) async {
    final originalAudit = await (select(biosecAudits)..where((a) => a.id.equals(auditId))).getSingle();

    final json = originalAudit.toJson();
    json[questionKey] = newValue;

    final newAudit = BiosecAudit.fromJson(json);

    return update(biosecAudits).replace(newAudit);
  }
```

and this works and I am mostly happy with it. However, the serialize/deserialize does feel a little clunky, and I thought I'd see if I missed a better way. How would you (anybody) do it?

(I did think about constructing a SQL statement, and SQL might be tidier. Would have to study on parameterizing the string values. Also I kind of like not getting into SQL if I can do it in a Drifty way. Still thinking this option over, comments welcome.)

**Comment by simolus3** [comment, marked as answer]
Is `questionKey` also the name of the column in SQL? If so, you can come pretty close to just writing SQL like this:

```dart
final stmt = update(biosecAudits)..where((a) => a.id.equals(auditId));
await stmt.write(RawValuesInsertable({questionKey: Variable(newValue)}));
```

This will run `UPDATE biosec_audits SET $questionKey = $newValue WHERE id = $auditId`, which looks like what you need?

**Reply by mattsimgithub** [reply]
No, `questionKey` is the camelCase name of the column, but the actual SQL column name is snake_case.

I was able to smooth this out with a `toSnakeCase` extension method, so:

```dart
    await stmt.write(RawValuesInsertable({questionKey.toSnakeCase: Variable(newValue)}));
```

And I do prefer this to the ser/deser I was doing before.

Drift is doing its own snake_case conversion somewhere, is that exposed in the library anywhere? I'd rather use the same one you're using if possible.

_(reply edited: earlier I said that `questionKey` was the direct SQL column name, but I was mistaken)_

**Reply by simolus3** [reply]
>Drift is doing its own snake_case conversion somewhere, is that exposed in the library anywhere

We're using the [recase](https://pub.dev/packages/recase) package for this.
